selectrum.el

```diff
@@ -808,7 +834,21 @@ into the user input area to start with."
   (interactive)
   (when selectrum--current-candidate-index
     (setq selectrum--current-candidate-index
-          (max (if selectrum--match-required-p 0 -1)
+          (max (cond ((and selectrum--match-required-p
+                           (not selectrum--default-candidate))
+                      ;; allow submitting the empty string
+                      -1)
+                     ((and (not selectrum--match-required-p)
+                           (or (not selectrum--default-candidate)
+                               (and selectrum--previous-input-string
+                                    (not (string-empty-p
+                                          selectrum--previous-input-string)))))
+                      ;; when match is not required but there is a default
+                      ;; candidate only allow selecting when prompt isn't
+                      ;; empty because the default candidate is already
+                      ;; selected anyway
```

Comment:
Can these comments be complete capitalized sentences?

Reply:
Done